src/js/utils/DSLUpdateUtil.js

```diff
@@ -0,0 +1,197 @@
+import Config from '../config/Config';
+import DSLExpression from '../structs/DSLExpression';
+import DSLFilterTypes from '../constants/DSLFilterTypes';
+import DSLUtil from './DSLUtil';
+
+const DSLUpdateUtil = {
+  /**
+   * Update the string representation of a node in order to match the new node
+   *
+   * @param {String} src - The source expression string
+   * @param {ASTNode} node - The ast node to replace
+   * @param {ASTNode} newNode - The ast node to replace with
+   * @param {number} offset - The offset to apply on position indices
+   * @returns {String} Returns the updated string
+   */
+  updateNodeString(src, node, newNode, offset=0) {
+    const {position, filterType} = node;
+    let {filterParams: {text, label}} = newNode;
+    if ((Config.environment === 'development') &&
+        (filterType !== newNode.filterType)) {
+      throw new Error('Trying to update a node with a mismatching node!');
+    }
+
+    switch (filterType) {
+      case DSLFilterTypes.ATTRIB:
+        // Attribute nodes require a special care since their value might be
+        // located in a different location than it's label (ex. multi-value)
+        //
+        // position[0] contains the `label:` part
+        // position[1] contains the `value` part
+        //
+        const labelStart = position[0][0] + offset;
+        const labelEnd = position[0][1] + offset;
+        const valueStart = position[1][0] + offset;
+        const valueEnd = position[1][1] + offset;
+
+        return src.substr(0, labelStart) + `${label}:` +
```

Comment:
It was kind of silly to represent the entire string with a template since I was adding more characters around the substr's making things more difficult to read.

The only place it actually did make sense was in the label, that's why the little template over there is on it's own.